openfisca_uk_data/datasets/frs/frs_enhanced/lcf_imputation.py

```diff
@@ -0,0 +1,228 @@
+        x_new=frs,
+    )
+
+
+def load_lcfs_with_carbon(year: int) -> MicroDataFrame:
+    """Load LCF data with carbon consumption.
+
+    Args:
+        year (int): The year of LCFS to use.
+
+    Returns:
+        MicroDataFrame: The LCF data with carbon consumption.
+    """
+    households, people, emissions = load_and_process_lcf(year)
+
+    # Manipulate LCF and NCFS data to get expenditure on the same categories
+    # Both datasets use the COICOP classification
+    # Documentation: https://unstats.un.org/unsd/classifications/unsdclassifications/COICOP_2018_-_pre-edited_white_cover_version_-_2018-12-26.pdf
+
+    index_to_col = {i: f"P6{i:02}" for i in CATEGORY_NAMES}
+    spending = (
+        households[list(index_to_col.values())]
+        .rename(columns={y: x for x, y in index_to_col.items()})
+        .unstack()
+        .reset_index()
+    )
+    spending.columns = "category", "household", "spending"
+    spending["household"] = households.case[spending.household].values
+    households = households.set_index("case")
+    spending.category = spending.category.map(CATEGORY_NAMES)
+    spending.spending *= 52
+    spending["weight"] = households.weighta[spending.household].values * 1000
+    spending = MicroDataFrame(spending, weights=spending.weight)
+    emissions.index = emissions.index.astype(str)
+    spending_by_category = spending.groupby("category").spending.sum()
+
+    grouped_ncfs = pd.DataFrame(
+        {
+            code: {
+                "category": CATEGORY_NAMES[code],
+                "carbon_tonnes": emissions[
+                    emissions.index.str.startswith(str(code))
+                ].carbon_tonnes.sum(),
+            }
+            for code in CATEGORY_NAMES.keys()
+        }
+    ).T.set_index("category")
+
+    # For each category, calculate spending from the LCFS and carbon emissions
+    # from the NCFS. Then, divide to find the carbon emissions per pound spent
+
+    carbon_by_category = pd.DataFrame(
+        {
+            category: {
+                "carbon_tonnes": grouped_ncfs.carbon_tonnes[category],
+                "spending": spending_by_category[category],
+            }
+            for category in spending.category.unique()
+        }
+    ).T
+
+    carbon_by_category["carbon_per_pound"] = (
+        carbon_by_category.carbon_tonnes / carbon_by_category.spending
+    )
+
+    # Multiple spending by carbon intensity to calculate LCF households' carbon footprints
+    spending["carbon_tonnes"] = (
+        carbon_by_category.carbon_per_pound[spending.category].values
+        * spending.spending
+    )
+    lcf_with_carbon = (
+        pd.DataFrame(spending[["household", "carbon_tonnes", "weight"]])
+        .groupby("household")
+        .sum()
+    )
+
+    # Add in LCF variables that also appear in the FRS-based microsimulation model
+
+    lcf_with_carbon = pd.concat(
+        [
+            lcf_with_carbon,
+            households[list(HOUSEHOLD_LCF_RENAMES.keys())].rename(
```

Comment:
Could you make this and the below `people` arg separate variables before the `concat`?

Reply:
Yep, added